### ISSUE TITLE
Broken Update Compendium 

### ISSUE THREAD

**Issue by TarionT** [earlSt1/vtt-compendium-folders#101]
**Describe the bug**
I have several actor comendiums with folders but beginning with version 2.3.5 i have a message that open the actor comepdium an say updatening folder structur and the compendium overlay will closed imediatly after that message apears.

The thing is this only happends on actor compendiums, all other comendiums scenes and journalitem works like a charm.

The error message in the console:
fic-folders.js:10 Uncaught (in promise) TypeError: Cannot read property 'update' of undefined
    at Function.packUpdateEntity (fic-folders.js:10)
    at async fic-folders.js:647

**To Reproduce**
it happend al the time even with no moduls expect comepndium folders and lib... activ

**Information**
- FoundryVTT Version: 0.8.8
- Compendium Folders Version: 2.3.5
- Game System + Version: TDE 2.2.0
- All other modules disabled? (y/n): y

**Comment by earlSt1** [comment]
Thanks this has been mentioned by other users, i'll push out a fix later this evening

**Comment by earlSt1** [comment]
Fixed in v2.3.51

**Comment by sneat** [comment]
I think this needs to be reopened @earlSt1 - the issue is System dependent (note that dnd5e is fine).

Running v2.3.51
System: DSA/TDE (the dark eye)

Exporting folder structure (shown below) to an empty compendium (world or module) results in:

```
Uncaught (in promise) TypeError: Cannot read property 'update' of undefined
    at Function.packUpdateEntity (fic-folders.js:10)
    at async fic-folders.js:647
```

I think there are two issues here:

1. The `#[CF_tempEntity]` actor has no data flags attached to it when it gets added to the compendium.

![image](https://user-images.githubusercontent.com/293277/129852748-7373ea9e-39aa-4bab-80d5-1eaa25d04e9a.png)
![image](https://user-images.githubusercontent.com/293277/129852836-c40a97e7-e2b9-482c-99e1-82e18de025c8.png)

2. The `entity` that gets passed into `removeOrUpdateFolderIdForEntity` https://github.com/earlSt1/vtt-compendium-folders/blob/082-update/fic-folders.js#L895 has a field called `_id` rather than `id` so when it gets passed back to the `updateData`, it ends up trying to update an undefined entity (because the id is `undefined`).

**Comment by earlSt1** [comment]
![image](https://user-images.githubusercontent.com/13738280/129867470-125cc3e3-8ec7-42e0-a915-9bd4c2a17f27.png)
Looks like the system is causing issues, it's clearing all flags when you create an actor (which wipes all CF folder data). I'll raise it with the systems guys

**Comment by Weilef** [comment]
![Error message](https://user-images.githubusercontent.com/89465686/130630340-33376c38-8019-46ea-98ab-1459d13dfae3.jpg)
With only the compendium folder module enabled and attempting to export folder structure, this happens. I have attempted it with only a single folder or more, and the same error occurs.

**Comment by earlSt1** [comment]
> ![Error message](https://user-images.githubusercontent.com/89465686/130630340-33376c38-8019-46ea-98ab-1459d13dfae3.jpg)
> With only the compendium folder module enabled and attempting to export folder structure, this happens. I have attempted it with only a single folder or more, and the same error occurs.
> 
> Nothing within the compendium appears to update when this happens.

Hi, which system + version are you using?

**Comment by Weilef** [comment]
Version 8.9 of Foundry with the latest version of the module with no other module running. I'm not sure if this completely resolves the issue (it looks like it so far) but I put an if(document) statement right where the undefined variable error message was located. It took away the error message and compendium was exporting again.

if(document) return await document.update....

**Comment by earlSt1** [comment]
Aha, I think I've identified the issue, was passing the incorrect id into that function. 

Can you try installing the test version of CF to see if it fixes it for you? To do so, uninstall CF and install this version using the manifest input at the bottom of the install screen:
https://raw.githubusercontent.com/earlSt1/vtt-compendium-folders/082-update/test/module.json

**Comment by Weilef** [comment]
> Aha, I think I've identified the issue, was passing the incorrect id into that function.
> 
> Can you try installing the test version of CF to see if it fixes it for you? To do so, uninstall CF and install this version using the manifest input at the bottom of the install screen:
> https://raw.githubusercontent.com/earlSt1/vtt-compendium-folders/082-update/test/module.json

I receive this error message.

fic-folders.js:1136 Uncaught (in promise) TypeError: Cannot read property 'flags' of undefined
    at Function.exportSingleFolderToCompendium (fic-folders.js:1136)
    at async Function.recursivelyExportFolders (fic-folders.js:1103)
    at async Function.recursivelyExportFolders (fic-folders.js:1117)
    at async callback (fic-folders.js:1016)

----------------------
        if (existingFolder){
            // Use existing
            folderId = existingFolder.data.flags.cf.id
            path = existingFolder.data.flags.cf.path  (X)

**Comment by earlSt1** [comment]
Ahhhhh, the manifest is wrong, I must have accidentally merged it in from the other branch

Can you try and install again using the same methods?

**Comment by Weilef** [comment]
> Ahhhhh, the manifest is wrong, I must have accidentally merged it in from the other branch
> 
> Can you try and install again using the same methods?

It's working. Thank you for the quick fix!

**Comment by earlSt1** [comment]
Talked through the issue with @TarionT, issue is now fixed in v2.3.53
